IdentityCore/src/requests/sdk/msal/MSIDDefaultBrokerResponseHandler.m

```diff
@@ -83,6 +87,17 @@ - (MSIDBrokerResponse *)brokerResponseFromEncryptedQueryParams:(NSDictionary *)e
         return nil;
     }
     
+    // In case MSAL requests AT POP but Broker response Bearer, MSAL returns error to user
+    if ([authScheme isMemberOfClass:MSIDAuthenticationSchemePop.class])
+    {
+        NSString *tokenType = [decryptedResponse msidObjectForKey:MSID_OAUTH2_TOKEN_TYPE ofClass:[NSString class]];
```

Comment:
would be great to see a unit test for this logic.

Reply:
Addressed. I will update in the next commit.